File: Python/06-旋转数组的最小数字.py
class Solution:
    def minNumberInRotateArray(self, rotateArray):
        # write code here
        size = len(rotateArray)
        if size == 0:
            return 0
        if size == 1:
            return rotateArray[0]
        res = rotateArray[0]
        for i in range(size - 1):
            if rotateArray[i] > rotateArray[i + 1]:
                return rotateArray[i + 1]
        return res

File: C++/06-旋转数组的最小数字.cpp
class Solution {
public:
    int minNumberInRotateArray(vector<int> rotateArray) {
        int len = rotateArray.size();
        if(len == 0)
            return 0;
        if(len == 1)
            return rotateArray[0];
        int res = rotateArray[0];
        for(int i = 0; i < len - 1; i++){
            if(rotateArray[i] > rotateArray[i + 1])
                return rotateArray[i + 1];
        }
        return res;
    }
};
